### PR TITLE
filter duplicated direct reference codes in dropdown

### PR DESCRIPTION
**Pull request by lukoktonos**
Direct reference codes were being duplicated in this dropdown:
image
They no longer are:
image

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1179
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
     * See [front-end testing instructions](https://github.com/projecttacoma/bonnie/wiki/Testing#frontend-testing) for getting coverage for front-end tests
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: N/A if using JIRA test for reasons listed below
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links:  NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @mayerm94
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
JIRA test passes locally. Simple change. Approved

**Review by alexanderelliott121 — DISMISSED** [review]

**Comment by alexanderelliott121** [comment]
My only concern is that there are now 3 distinct places where codes are checked against /-/ to see if they are direct reference.

I could definitely see it being desirable to create a utility function to perform the check so that it is centralized.

**Review by alexanderelliott121 — DISMISSED** [review]

File: app/assets/javascripts/views/patient_builder/data_criteria.js.coffee [hunk 1]
```diff
@@ -96,21 +96,21 @@ class Thorax.Views.EditCriteriaView extends Thorax.Views.BuilderChildView
         for item in model.attributes.values
           # Add OR logic for any collections that need to display dates here
           if item.type == 'FAC'
-            start_date: moment.utc(item.value).format('L')    
-            start_time: moment.utc(item.value).format('LT')   
-            end_date: moment.utc(item.end_value).format('L') 
+            start_date: moment.utc(item.value).format('L')
+            start_time: moment.utc(item.value).format('LT')
+            end_date: moment.utc(item.end_value).format('L')
```
Comment (jbradl11) [inline app/assets/javascripts/views/patient_builder/data_criteria.js.coffee]: Could you reverse these white space changes.

File: spec/javascripts/fixtures/json/measure_data/CQL/CMS107/CMS107v6.json [hunk 2]
```diff
@@ -14947,19 +14947,19 @@
       "version": "N/A"
     },
     {
-      "oid": "1972a18c-9fd0-4845-8022-1b24cae9c658",
```
Comment (mayerm94) [inline spec/javascripts/fixtures/json/measure_data/CQL/CMS107/CMS107v6.json:14950]: What was the motivation behind changing these value set oids?

Comment (lukoktonos) [reply spec/javascripts/fixtures/json/measure_data/CQL/CMS107/CMS107v6.json:14950]: The oids in the measure.json didn't match the oids in the value_sets.json file:

image

This meant that they triggered the 'WARNING: missing value set' message in `Measure.valuesets()`:
image

I believe it's an issue with the fixture export tool.  I made a JIRA ticket to investigate (https://jira.mitre.org/browse/BONNIE-1208)

**Review by mayerm94 — APPROVED** [review]

**Review by jbradl11 — APPROVED** [review]
Code looks good. Tests pass. Manually confirmed that direct reference codes still work as intended.

**Pull request merged**